custom_components/opnsense/diagnostics.py

```
@@ -0,0 +1,334 @@
+            tuple(f"_{item}" for item in _URL_FIELDS)
+        ):
+            return "url"
+        if field_name in _ID_FIELDS or field_name.endswith(
+            tuple(f"_{item}" for item in _ID_FIELDS)
+        ):
+            return "id"
+        if field_name in {"email", "user"}:
+            return "user"
+        return "value"
+
+    def _replace_scalar(self, value: Any) -> Any:
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)